crates/services/p2p/src/service.rs

```diff
@@ -1691,6 +1728,9 @@ pub mod tests {
             heartbeat_max_time_since_last,
             next_check_time: Instant::now(),
             heartbeat_peer_reputation_config: heartbeat_peer_reputation_config.clone(),
+            cached_view: Arc::new(CachedView::new(false)),
+            cache_reset_interval: Duration::from_secs(0),
+            next_cache_reset_time: Instant::now(),
```

Comment:
I see that this is handled in the same way as pre-existing [heartbeat reputation check](https://github.com/FuelLabs/fuel-core/blob/master/crates/services/p2p/src/service.rs#L963-L973), but I'm wondering if we should allow `0` as the interval.

Is there a risk that the cache will never be cleared?

Reply:
in https://github.com/FuelLabs/fuel-core/blob/f597ac9db0602caed7e20eaeafed0a1bd2e96ca7/crates/services/p2p/src/service.rs#L828 we define it to be 10sec, we set it to 0 just in the tests for reduced complexity.
the cache *may* not be cleared since it is the least priority future in the `tokio::select` expansion ~ but it will eventually happen if it was missed.

Reply:
good point though, had to reorder the priorities to ensure it clears :)